File: backend/src/main/java/edu/cit/cabatana/doughlycrumbl/model/Order.java
package edu.cit.cabatana.doughlycrumbl.model;

import edu.cit.cabatana.doughlycrumbl.features.user.User;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "orders")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false, length = 60)
    @Builder.Default
    private String status = "PENDING";

    // Payment status — default "UNPAID" so Hibernate never inserts null.
    // columnDefinition also sets a DB-level DEFAULT so existing rows and
    // any raw SQL inserts that omit the column stay consistent.
    @Column(name = "payment_status", nullable = false, length = 30,
            columnDefinition = "VARCHAR(30) DEFAULT 'UNPAID'")
    @Builder.Default
    private String paymentStatus = "UNPAID";

    @Column(name = "delivery_address", nullable = false, columnDefinition = "TEXT")
    private String deliveryAddress;

    @Column(name = "contact_number", nullable = false, length = 20)
    private String contactNumber;

    @Column(name = "delivery_notes", columnDefinition = "TEXT")
    private String deliveryNotes;

    @Column(name = "proof_image_url", length = 500)
    private String proofImageUrl;

    @Column(name = "cancellation_reason", columnDefinition = "TEXT")
    private String cancellationReason;

    @Column(name = "total_amount", nullable = false, precision = 10, scale = 2)
    private BigDecimal totalAmount;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<OrderItem> items = new ArrayList<>();

    @CreationTimestamp
    @Column(name = "order_date", nullable = false, updatable = false)
    private LocalDateTime orderDate;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}

File: mobile/app/src/main/java/com/example/mobile/model/Order.kt
package com.example.mobile.model

import com.google.gson.annotations.SerializedName

data class Order(
    @SerializedName("orderId")
    val orderId: Long = 0,

    @SerializedName("orderDate")
    val orderDate: String = "",

    @SerializedName("status")
    val status: String = "",

    @SerializedName("paymentStatus")
    val paymentStatus: String = "",

    @SerializedName("deliveryAddress")
    val deliveryAddress: String? = null,

    @SerializedName("contactNumber")
    val contactNumber: String? = null,

    @SerializedName("deliveryNotes")
    val deliveryNotes: String? = null,

    @SerializedName("fulfillmentMethod")
    val fulfillmentMethod: String? = null,

    @SerializedName("paymentMethod")
    val paymentMethod: String? = null,

    @SerializedName("proofImageUrl")
    val proofImageUrl: String? = null,

    @SerializedName("cancellationReason")
    val cancellationReason: String? = null,

    @SerializedName("items")
    val items: List<OrderItem> = emptyList(),

    @SerializedName("subtotalAmount")
    val subtotalAmount: Double? = null,

    @SerializedName("deliveryFee")
    val deliveryFee: Double? = null,

    @SerializedName("totalAmount")
    val totalAmount: Double = 0.0,

    @SerializedName("itemCount")
    val itemCount: Int? = null
)
